### PR TITLE
1.0.10

### PR DESCRIPTION
**Pull request by andriycraft**
title

### CODE REVIEW

**Comment by andriycraft** [comment]
Log:
[INFO] Running tests
[INFO] Config validator

**Comment by andriycraft** [comment]
Invalid config:
[23:40:43] INFO > Loading Server...
[23:40:43] INFO > Loading BlueBird.json
[23:40:43] INFO > Loading Lang.json
[23:40:43] INFO > This server is running BlueBird, v1.0.10
[23:40:43] INFO > BlueBird is distributed under GPLv3 License
[23:40:43] WARNING > [CONFIG ERROR] Your config is outdated!
[23:40:43] WARNING > [CONFIG ERROR] missing 'etc' category
[23:40:43] CRITICAL > Unable to start server.

**Comment by andriycraft** [comment]
Valid config

**Comment by andriycraft** [comment]
all tests have failed

**Comment by VaxPex** [comment]
??????

**Comment by andriycraft** [comment]
> ??????

?

**Comment by andriycraft** [comment]
this is the biggest pr i ever made

**Comment by andriycraft** [comment]
misclicked

**Comment by andriycraft** [comment]
TODO:
[] Commands
[] Remove debug

**Comment by andriycraft** [comment]
debugging this is pain

**Comment by andriycraft** [comment]
WARNING: This pr is not ready to be merged

**Comment by andriycraft** [comment]
@VaxPex how to read packet in bluebird?

**Comment by andriycraft** [comment]
@VaxPex are you alive?

**Comment by andriycraft** [comment]
it works

**Comment by andriycraft** [comment]
@VaxPex please merge this pr and give me contributor role on discord

**Comment by VaxPex** [comment]
> @VaxPex please merge this pr and give me contributor role on discord

fix the disconnection then

**Comment by andriycraft** [comment]
> > @VaxPex please merge this pr and give me contributor role on discord
> 
> fix the disconnection then



> > @VaxPex please merge this pr and give me contributor role on discord
> 
> fix the disconnection then

What disconnection? It works

**Comment by VaxPex** [comment]
foreach the cmds then cmdidkwhat.dispatch(handler.player, cmd.substring(1))

**Comment by MyrsujV2** [comment]
hi

**Comment by MyrsujV2** [comment]
does it work

**Comment by MyrsujV2** [comment]
or no

**Comment by MyrsujV2** [comment]
or yes

**Comment by MyrsujV2** [comment]
or no

**Comment by MyrsujV2** [comment]
or yes

**Comment by andriycraft** [comment]
yes

**Comment by AnyBananaGAME** [comment]
ok!

**Comment by AnyBananaGAME** [comment]
https://tenor.com/view/deadserver-gif-22763122

**Comment by andriycraft** [comment]
> https://tenor.com/view/deadserver-gif-22763122

no

**Comment by andriycraft** [comment]
ik its rickroll

**Comment by andriycraft** [comment]
@VaxPex it works
will u merge it

**Comment by andriycraft** [comment]
?

**Comment by andriycraft** [comment]
no vaxpex dont merge it

**Comment by andriycraft** [comment]
it contains bugs

**Comment by MyrsujV2** [comment]
it works ig

**Review by PiggyPlex — APPROVED** [review]

**Review by PiggyPlex — COMMENTED** [review]
seems sensible

**Review by PiggyPlex — APPROVED** [review]

**Review by PiggyPlex — APPROVED** [review]

File: src/Server.js [hunk 1]
```diff
@@ -91,6 +91,9 @@ class Server {
 					"port": 19132,
 					"version": 4,
 				},
+				"etc": {
+					"gamemode": ""
```
Comment (PiggyPlex) [inline src/Server.js]: should we not set a default?

File: src/Server.js [hunk 2]
```diff
@@ -122,11 +125,25 @@ class Server {
 		let addrname = this.bluebirdcfg.getNested("address.name");
 		let addrport = this.bluebirdcfg.getNested("address.port");
 		let addrversion = this.bluebirdcfg.getNested("address.version");
-		this.raknet = new RakNetHandler(this, addrname, addrport, addrversion);
+		try {
+			this.raknet = new RakNetHandler(this, addrname, addrport, addrversion);
+		} catch (e) {
```
Comment (PiggyPlex) [inline src/Server.js:127]: unused variable `e`. consider logging the error to the console or prefixing it with `_`.

File: src/Server.js [hunk 3]
```diff
@@ -122,11 +125,25 @@ class Server {
...
+			this.getLogger().critical(`Raknet failed to start. Is the address configured correctly?`)
+			this.getLogger().critical(`Unable to start server.`)
+			process.exit(1)
+		}
 		if (this.raknet.raknet.isRunning === true) {
 			this.raknet.handle();
 		}
 
+		if (!this.bluebirdcfg.getNested("etc.gamemode")) {
```
Comment (PiggyPlex) [inline src/Server.js]: should use a schema validator instead

**Review by PiggyPlex — CHANGES_REQUESTED** [review]

**Review by PiggyPlex — APPROVED** [review]

**Review by PiggyPlex — APPROVED** [review]

**Comment by MyrsujV2** [comment]
@PiggyPlex You approved this pr 4 times

**Comment by VaxPex** [comment]
😂

**Comment by AnyBananaGAME** [comment]
If all of the kings had their queens on the throne
We would pop champagne and raise a toast
To all of the queens who are fighting alone
Baby, you're not dancin' on your own
Can't live without me, you wanna, but you can't, no, no, no
Think it's funny, but honey, can't run this show on your own
I can feel my body shake, there's only so much I can take
I'll show you how a real queen behaves, oh
No damsel in distress, don't need to save me
Once I start breathin' fire, you can't tame me
And you might think I'm weak without a sword
But if I had one, it'd be bigger than yours
If all of the kings had their queens on the throne
We would pop champagne and raise a toast
To all of the queens who are fighting alone
Baby, you're not dancin' on your own
Disobey me, then baby, it's off with your head
Gonna change it and make it a world you won't forget, oh-oh, oh
No damsel in distress, don't need to save me
Once I start breathin' fire, you can't tame me
And you might think I'm weak without a sword
But I'm stronger than I ever was before
If all of the kings had their queens on the throne
We would pop champagne and raise a toast
To all of the queens who are fighting alone
Baby, you're not dancin' on your own
In chess, the king can move one space at a time
But queens are free to go wherever they like
You get too close, you'll get a royalty high
So breathe it in to feel alive (alive)
If all of the kings had their queens on the throne
We would pop champagne and raise a toast
To all of the queens who are fighting alone
Baby, you're not dancin' on your own

**Pull request closed without merging**